builtin/logical/database/backend.go

```diff
@@ -228,6 +272,156 @@ func (b *databaseBackend) GetConnection(ctx context.Context, s logical.Storage,
 	return db, nil
 }
 
+// initQueue preforms the necessary checks and initializations needed to preform
+// automatic credential rotation for roles associated with static accounts. This
+// method verifies if a queue is needed (primary server or local mount), and if
+// so initializes the queue and launches a go-routine to periodically invoke a
+// method to preform the rotations.
+//
+// initQueue is invoked by the Factory method in a go-routine. The Factory does
+// not wait for success or failure of it's tasks before continuing. This is to
+// avoid blocking the mount process while loading and evaluating existing roles,
+// etc.
+func (b *databaseBackend) initQueue(ctx context.Context, conf *logical.BackendConfig) {
+	// verify this mount is on the primary server, or is a local mount. If not, do
+	// not create a queue or launch a ticker. Both processing the WAL list and
+	// populating the queue are done sequentially and before launching a
+	// go-routine to run the periodic ticker.
+	replicationState := conf.System.ReplicationState()
+	if (conf.System.LocalMount() || !replicationState.HasState(consts.ReplicationPerformanceSecondary)) &&
```

Comment:
I don't think this is right.  For example if you are a local mount running on a performance secondary, I don't think you want to start this queue.

Reply:
cc @briankassouf can you confirm?

Reply:
A local mount on a performance secondary should start the queue. I believe this logic is correct, but i think it might be more straightforward as a switch statement. Something like:
```
switch {
case replicationState.HasState(consts.ReplicationDRSecondary),
         replicationState.HasState(consts.ReplicationPerformanceStandby),
         !conf.System.LocalMount() && replicationState.HasState(consts.ReplicationPerformanceSecondary:
               return
}
```

Reply:
Mostly just inversing the logic and making it OR is IMHO cleaner

Reply:
I added a task to our internal tracker to pick this up as a follow-up after this PR, sound good?